src/lib.rs

```diff
@@ -254,6 +246,21 @@ impl Shape {
         }
     }
 
+    pub fn indented(config: &Config, indent: Indent) -> Shape {
+        Shape::legacy(config.max_width.checked_sub(indent.width()).unwrap_or(0),
+                      indent)
```

Comment:
Do we save much code by using the `legacy` ctor vs init'ing `Shape` directly? I'd prefer the latter unless there is a big saving in code complexity